modules/edit-modes/src/lib/transform-mode.js

```diff
@@ -0,0 +1,319 @@
+// @flow
+
+import envelope from '@turf/envelope';
+import { point, featureCollection, lineString } from '@turf/helpers';
+import { coordEach } from '@turf/meta';
+import { getCoord, getGeom } from '@turf/invariant';
+import turfBuffer from '@turf/buffer';
+import turfCentroid from '@turf/centroid';
+import turfDistance from '@turf/distance';
+import turfBearing from '@turf/bearing';
+import turfTransformTranslate from '@turf/transform-translate';
+import turfTransformScale from '@turf/transform-scale';
+import turfTransformRotate from '@turf/transform-rotate';
+import { getPickedEditHandle } from '../utils';
+import type {
+  PointerMoveEvent,
+  StartDraggingEvent,
+  StopDraggingEvent,
+  DraggingEvent,
+  ModeProps,
+  EditHandleFeature
+} from '../types.js';
+import type { FeatureCollection, Position, Feature } from '../geojson-types.js';
+import { BaseGeoJsonEditMode, getIntermediatePosition } from './geojson-edit-mode';
+import { ImmutableFeatureCollection } from './immutable-feature-collection';
+
+export class TransformMode extends BaseGeoJsonEditMode {
+  _cornerGuidePoints: Array<EditHandleFeature> = [];
+  _previousMouseCoords: ?Position;
+  _transformMode: ?string;
+  _geometryBeforeTransform: ?FeatureCollection;
+  _selectedEditHandle: ?EditHandleFeature;
+  _cursor: ?string;
+
+  _getOppositeScaleHandle = (selectedHandle: EditHandleFeature) => {
+    const selectedHandleIndex =
+      selectedHandle && selectedHandle.properties && selectedHandle.properties.index;
+    if (typeof selectedHandleIndex !== 'number') return null;
+    const guidePointCount = this._cornerGuidePoints.length;
+    const oppositeIndex = (selectedHandleIndex + guidePointCount / 2) % guidePointCount;
+    return this._cornerGuidePoints.find(p => p.properties.index === oppositeIndex);
+  };
+
+  _getUpdatedData = (props: ModeProps<FeatureCollection>, editedData: FeatureCollection) => {
+    let updatedData = new ImmutableFeatureCollection(props.data);
+    const selectedIndexes = props.selectedIndexes;
+    for (let i = 0; i < selectedIndexes.length; i++) {
+      const selectedIndex = selectedIndexes[i];
+      const movedFeature = editedData.features[i];
+      updatedData = updatedData.replaceGeometry(selectedIndex, movedFeature.geometry);
+    }
+    return updatedData.getObject();
+  };
+
+  _isSinglePointSelected = (): boolean => {
+    const { features } = this._geometryBeforeTransform || {};
+    if (Array.isArray(features) && features.length === 1) {
+      const { type } = getGeom(features[0]);
+      if (type === 'Point') {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  _getTranslateAction(
```

Comment:
Is there a way for this mode to use the translate, scale, and rotate modes inside itself rather than duplicating their logic?

Reply:
I updated my PR - scale and rotate modes now have edit handle guides. Translate mode functionality will remain unchanged. Transform mode's functionality will remain the same as what's depicted in GIF attached to this PR's description.